server/utils/passport/ensure-admin.js

```diff
@@ -8,9 +8,13 @@ export default async function ensureAdmin(req, res, next) {
     const { appDb } = req.app.locals
     const query = { uid: req.user, role: 'admin' }
     const user = await UserModel.findOne(appDb, query)
+    const isOwnAdminAccount = req?.params?.uid === user?.uid
 
     if (!user) return res.status(401).json({ error: 'Incorrect access level.' })
 
+    if (isOwnAdminAccount)
```

Comment:
Sorry, I think I was a little unclear. My understanding is all users should be able to edit their own accounts, admin or not. I think otherwise we break the profile page. I was thinking something else in `UsersController.edit` that looks like this:
```javascript
const isSelfOrAdmin = user.role === 'admin' || user.uid === req.user

if (!isSelfOrAdmin) {
  return res.status(401)
}
```